src/Compilers/VisualBasic/Test/Syntax/Parser/ParseDirectives.vb

```diff
@@ -22,6 +22,20 @@ Public Class ParseDirectives
     End Sub
 
     <Fact>
+    Public Sub ParseReferenceDirective()
+        ParseAndVerify(<![CDATA[
+            #r "reference"
```

Comment:
If we still support `#reference` (I've lost track), there should be a test for that as well.